pkg/controllers/edgeconnect/deployment/deployment.go

```diff
@@ -156,21 +116,55 @@ func edgeConnectContainer(instance *edgeconnectv1alpha1.EdgeConnect, resource st
 			RunAsUser:                address.Of(unprivilegedUser),
 			RunAsNonRoot:             address.Of(true),
 		},
-		VolumeMounts: []corev1.VolumeMount{
-			{MountPath: consts.EdgeConnectMountPath, Name: consts.EdgeConnectVolumeMountName, ReadOnly: true},
-		},
+		VolumeMounts: prepareVolumeMounts(instance),
 	}
 }
 
-func prepareVolume(clientSecretName string) corev1.Volume {
+func prepareVolumes(instance *edgeconnectv1alpha1.EdgeConnect) []corev1.Volume {
+	volumes := []corev1.Volume{prepareConfigVolume()}
+
+	if instance.Spec.CaCertsRef != "" {
+		volumes = append(volumes, corev1.Volume{
+			Name: consts.EdgeConnectCustomCAVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: instance.Spec.CaCertsRef,
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "certs",
```

Comment:
"magic" string?
maybe a `const` would be better? 🤔

Reply:
hmmm i think i fixed it